### PR TITLE
New changes made 

### PR DESCRIPTION
**Pull request by braincodeishan**
Kindly review and merge

### CODE REVIEW

File: src/components/ProductDetail.jsx [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
-    </div>
+    <Container fluid>
+      {isLoading && <Box>Loading</Box>}
+      {!isLoading && !error && (
+      <Grid>
+        <Grid.Col span={6}>
+          <Image src={productData.image} height={'auto'} width={'80%'} alt="ProductImage" />
+        </Grid.Col>
+        <Grid.Col span={6}>
+            dsghajdghsgh
```
Comment (jishnukoottala) [inline src/components/ProductDetail.jsx]: remove this "dsghajdghsgh"

File: src/components/ProductDetail.jsx [hunk 2]
```diff
@@ -1,9 +1,29 @@
-import { Link } from "react-router-dom";
+import React from 'react'
+import { Container, Grid, Title, Image, Text,Box } from "@mantine/core";
+import { useParams } from "react-router"
+import { useGetSingleProductQuery } from '../store/products/productsSlice';
 const ProductDetail = () => {
+  const {id}=useParams()
+  console.log(id);
+  const { data: productData, error, isLoading } = useGetSingleProductQuery(id);
+  console.log("product data",productData);
```
Comment (jishnukoottala) [inline src/components/ProductDetail.jsx]: remove console log

File: src/store/products/productsSlice.js [hunk 3]
```diff
@@ -8,9 +8,21 @@ export const productsApi = createApi({
...
     getProducts: builder.query({
-      query: () => `products`,
+      query: () => {
+        console.log("HIT ONCE and called api");
+        return `products`
+      },
+    }),
+    getSingleProduct: builder.query({
+      query: (id) => {
+        console.log("HIT ONCE and called api");
```
Comment (jishnukoottala) [inline src/store/products/productsSlice.js]: remove console log

File: src/store/products/productsSlice.js [hunk 4]
```diff
@@ -8,9 +8,21 @@ export const productsApi = createApi({
...
+        return `products/${id}`;
+      },
     }),
   }),
 });
 
-export const { useGetProductsQuery } = productsApi;
+console.log(productsApi);
```
Comment (jishnukoottala) [inline src/store/products/productsSlice.js]: remove console log

File: src/components/Product.jsx [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
       <Card shadow="sm" padding="lg" radius="md" withBorder>
-        <Card.Section>
-          <Image src={productImage} height={160} alt="ProductImage" />
+        <Card.Section className="productImage" onClick={()=>navigate(`/product/${id}`)}>
+          <Image src={productImage} height={'400px'} width={'300px'} alt="ProductImage" />
         </Card.Section>
 
         <Group position="apart" mt="md" mb="xs">
-          <Text weight={500}>{title}</Text>
+          <Text weight={500}>{title.length>30?title.substring(0,29)+"...":title}</Text>
```
Comment (jishnukoottala) [inline src/components/Product.jsx]: set width/height of this text box, and and overflow: ellipsis

**Review by jishnukoottala — CHANGES_REQUESTED** [review]
please do all these changes

File: src/components/ProductDetail.jsx (hunk 2 above)
Comment (braincodeishan) [reply src/components/ProductDetail.jsx]: done

File: src/store/products/productsSlice.js (hunk 3 above)
Comment (braincodeishan) [reply src/store/products/productsSlice.js]: done

File: src/store/products/productsSlice.js (hunk 4 above)
Comment (braincodeishan) [reply src/store/products/productsSlice.js]: done

File: src/components/Product.jsx (hunk 5 above)
Comment (braincodeishan) [reply src/components/Product.jsx]: done

**Comment by braincodeishan** [comment]
all changes requested are fulfilled.

**Pull request merged**